### ISSUE TITLE
Redirect not working on IOS

### ISSUE THREAD

**Issue by aklt** [apache/cordova-plugin-inappbrowser#484]
Hello,

We have built an app that logs in via OpenId Connect. The login on our
Cordova app functions well on Android devices but fails when we log in on an
IOS device.  This seems to be because the redirects that happen between the
inAppBrowser and the ADFS server function differently on IOS than they do on
an Android device.

The sequence of requests are something linke this on both Android and IOS:

     POST credentials              -->    ADFS Server
      GET uriWithRedirct header    -->    ADFS Server
      GET redirectURI              -->     APP Backend

(In the chrome Network tab the POST is shown as cancelled and the next line is
a GET to the same URI where the response contains a "Location: ..." header)

On IOS this results in a blank page with the title "HTTP 1.1 Service
Unavailable" which I think is generated by IOS or in the inAppBrowser
somewhere as I am quite sure that our APP backend does not generate a page
with this content.

With this page showing it is now possible to log in by pressing the back
button on the IOS device.  This results in a correct login to the app backend
to which we have been redirected, but of course we are not interested in
having to manually press the back button to log in :)

On Android the login happens correctly without this page showing and I suspect
that there is something going wrong with the redirects, but I am not sure how
to solve this.

Has anyone run into the same or a issue?

Note that we are not using the wkwebview.

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by janpio** [comment]
Can you build a recreation app for this? A plain Cordova app with a button that opens an InAppBrowser to a website that does the same redirections as your login process should be able to showcase what is goinh wrong exactly.

**Comment by aklt** [comment]
I agree that an app that reproduces the issue along with an OpenId Connect server to test against would be very useful to track down the problem and I would like to have a look at that when I find the time.

The first thing I would like to try is to use the wkwebview, maybe that behaves better on IOS with these redirects, but I will let you know when I have an update.

**Comment by janpio** [comment]
I think you can probably skip the OpenID Connect server, a simple web page that executes the same redirect pattern should have the same effect if this is indeed a problem with InAppbrowser.

**Comment by aklt** [comment]
Yes, but at least some backend that sends the same redirect as the OpenId Connect  server would be needed I think.  Unfortunately I have not been able to see exactly what is sent over the network.  The cancelled post which seems to turn into a GET above puzzles me.

I imagine it could be a problem with how communication with the Safari webview is done, and that this behavior is different to how the InAppBrowser communicates with Chrome or which browser is used for the webview there.

If I open a Safari on my iPhone and try to log in towards the OpenId Connect server I am redirected correctly without seeing "HTTP 1.1 Service Unavailable" page.

**Comment by janpio** [comment]
I would use the working Android app and some web debugging proxy like Charles or even just a remote debugging session with Chrome to understand the redirection chain and set up a few super simple PHP scripts that reproduce the same redirects. Maybe that is enough to already trigger the problem in an iOS app with InAppBrowser. 

If not, there is probably something going on with headers (also add that to the PHP mock) or security blocking (look at what is being logged in console or native logs more).

**Comment by aklt** [comment]
Thanks for your input.  The idea of using a mock server I really like. I did start going down that route using Wireshark, but ran into difficulties in getting the right setup proxying my phone towards the OpenId Connect ADFS server and dropped it in favor of trying to use the wkwebview.

And then other work got in my way, but I hope to be able to make this simple setup and app that cen replicate the problem.  That is, if the wkwebview does not solve the issue.  I will update here when I know more.

**Comment by janpio** [comment]
https://www.charlesproxy.com/ is a godsend in these cases, or my instructions at  https://ionic.zone/debug/remote-debug-your-app which allows you do look at the network panel of Safari/Chrome.

**Comment by aklt** [comment]
I have found a way to work around this issue with a hack:  If the page with the "HTTP 1.1 Service
Unavailable" message is shown some code in the InAppBRowser is executed causing it to go back to the actual page we want to show.

This works, but is most definitely a hack and it would be very nice to solve this in the right way.  I did not yet find time to test this using the wkwebview and I am not sure I will in the near future.
